packages/wallet/Account.js

```diff
@@ -0,0 +1,604 @@
+   */
+  signAuthenticationChallenge(transaction) {
+    // build the transaction
+    const txEnvelope = StellarSdk.xdr.TransactionEnvelope.fromXDR(transaction, 'base64');
+    const tx1 = new StellarSdk.Transaction(txEnvelope);
+    // sign it
+    tx1.sign(this.account);
+    return tx1.toEnvelope().toXDR('base64');
+  }
+
+  /**
+   * Send the signed transaction XDR to the server, which is saved before
+   * @param {string} signedTransaction - the XDR encoded signed transaction
+   * @returns {object|null}the JWT token, if verification on the server is successful
+   * otherwise null
+   */
+  async sendSignedAuthentication(signedTransaction) {
+    try {
+      const options = {
+        method: 'POST',
+        uri: this.webAuthEndpoint,
+        body: {
+          transaction: signedTransaction
+        },
+        json: true // Automatically stringifies the body to JSON
+      };
+      const response = await rp(options);
+      return response;
+    } catch (error) {
+      console.log(error);
+      return null;
+    }
+  }
+
+  /**
+   * Return the network where the account is.
+   * @returns the network where the account is
+   */
+  getNetwork() {
+    return this.network;
+  }
+
+  /**
+   * Get the balance associated to this account.
+   * Currently only values for XLM and ETH are saved.
+   * i.e. there is no other types of currency.
+   * @returns {string} the balance associated to this account in string format.
+   */
+  getBalance() {
+    return this.balance;
+  }
+
+  /**
+   * Get transaction history in and out from this account.
+   * Currently it is the raw json response, and
+   * different between eth and stellar.
+   * @returns {JSON} transaction history in and out from this account
+   */
+  getHistory() {
+    return this.history;
+  }
+
+  /**
+   * Get the account that is passed into this wrapper class.
+   * @returns {object} the account that is passed into this wrapper class
+   */
+  getOriginalAccount() {
+    return this.account;
+  }
+
+  /**
+   * The address of a stellar account is the public key of the key pair;
+   * The address of an ethereum account is a part of the hash of the public key
+   * @returns {string} the address
+   */
+  getAddress() {
+    switch (this.network) {
+      case 'stellar':
+        return this.account.publicKey();
+      case 'ethereum':
+        return this.account.address;
+      default:
+        console.log('The network is not set correctly.');
+        return null;
+    }
+  }
+
+  /**
+   * Return the private key (ethereum) / secret (stellar)
+   * @returns {string} the private key
+   */
+  getPrivateKey() {
+    switch (this.network) {
+      case 'stellar':
+        return this.account.secret();
+      case 'ethereum':
+        return this.account.privateKey;
+      default:
+        console.log('The network is not set correctly.');
+        return null;
+    }
+  }
+
+  // Phase 3 Anchor Bridge, client side implementation of SEP-0006
```

Comment:
Let's move Anchor Bridge out to another file (`Anchor.js`) for clearer separation of concerns.

Reply:
separated